### DISCUSSION TITLE
Add a button to close banner with rejection

### DISCUSSION THREAD

**Discussion by montelor** [orestbida/cookieconsent#308] [Q&A]
Thanks for this wonderful project.
I would like to use it, but I don't know how to add a close (and reject) button like "X" at the top right of the banner, as requested by the privacy guarantor.
I'm not a developer, but if it wasn't too complicated I would like to implement in the banner.
Can you give me the code please?
Thanks for your help and have a nice day.

**Comment by dombyte** [comment]
Hmm, you could use "accept_necessary" for the second button. I don't know if it`s technical possible to reject all cookies because you need to set at least one cookie to save your cookie settings. Even the reject all button in the dedicated settings tab creates the cc_cookie a cookie used to save your settings. I just would use this method and only use the cookie consent cookie as an technical necessary cookie. Also if you're using google fonts or different things which are necessary for your page to work probably you don't want to block all, otherwise your site brakes.

**Reply by montelor** [reply]
Thanks for the quick reply!!
True, it is necessary to use "accept_necessary", because certain cookies are necessary for the functioning of the page.
The problem is that I don't know how to add it to get a banner like this (example).
Let me explain, what code do I use and where do I put it, in the cookieconsent-init.js file or in the cookieconsent.js file?
I work fine with html but javascript is really hard ...
Could you please provide me with a snippet?
Thanks for your availability!

Montelor
![banner-x](https://user-images.githubusercontent.com/105868051/169332270-c65b222e-c276-4fde-b9fd-cb82f2c9156e.png)

**Reply by dombyte** [reply]
**Sure!**

**That's the cookieconsent-init.js File where you edit all your Personal settings. You can use this, there I have already changed it for you, this is also the iframe manager version so you can block YouTube etc. Im currently working on my Personal Blog where I also use this Plugin, I will make it available on git later this day or tomorrow so there you can maybe find also find some inspiration. 
It makes sense when you use the iframe manager to make a new cookie group because by default I think it uses the analytics one.**

```
var LOREM_IPSUM = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.';

// obtain iframemanager object
var manager = iframemanager();

// obtain cookieconsent plugin
var cc = initCookieConsent();

// Configure with youtube embed
manager.run({
    currLang: 'en',
    services : {
        youtube : {
            embedUrl: 'https://www.youtube-nocookie.com/embed/{data-id}',
            thumbnailUrl: 'https://i3.ytimg.com/vi/{data-id}/hqdefault.jpg',
            iframe : {
                allow : 'accelerometer; encrypted-media; gyroscope; picture-in-picture; fullscreen;',
            },
            cookie : {
                name : 'cc_youtube'
            },
            languages : {
                en : {
                    notice: 'This content is hosted by a third party. By showing the external content you accept the <a rel="noreferrer" href="https://www.youtube.com/t/terms" title="Terms and conditions" target="_blank">terms and conditions</a> of youtube.com.',
                    loadBtn: 'Load video',
                    loadAllBtn: 'Don\'t ask again'
                }
            }
        }
    }
});

// run plugin with config object
cc.run({
    current_lang: 'en',
    autoclear_cookies: true,                    // default: false
    cookie_name: 'cc_cookie_demo2',             // default: 'cc_cookie'
    cookie_expiration: 365,                     // default: 182
    // page_scripts: false,                         // default: false
    force_consent: true,                        // default: false

    // auto_language: null,                     // default: null; could also be 'browser' or 'document'
    // autorun: true,                           // default: true
    // delay: 0,                                // default: 0
    // hide_from_bots: false,                   // default: false
    // remove_cookie_tables: false              // default: false
    // cookie_domain: location.hostname,        // default: current domain
    // cookie_path: '/',                        // default: root
    // cookie_same_site: 'Lax',
    // use_rfc_cookie: false,                   // default: false
    // revision: 0,                             // default: 0

    gui_options: {
        consent_modal: {
            layout: 'box',                    // box,cloud,bar
            position: 'middle center',          // bottom,middle,top + left,right,center
            transition: 'slide'                 // zoom,slide
        },
        settings_modal: {
            layout: 'bar',                      // box,bar
            position: 'left',                   // right,left (available only if bar layout selected)
            transition: 'slide'                 // zoom,slide
        }
    },

    onFirstAction: function(){
        console.log('onFirstAction fired');
    },

    onAccept: function(){
        console.log('onAccept fired!')

        // If analytics category is disabled => load all iframes automatically
        if(cc.allowedCategory('analytics')){
            console.log('iframemanager: loading all iframes');
            manager.acceptService('all');
        }
    },

    onChange: function (cookie, changed_preferences) {
        console.log('onChange fired!');

        // If analytics category is disabled => ask for permission to load iframes
        if(!cc.allowedCategory('analytics')){
            console.log('iframemanager: disabling all iframes');
            manager.rejectService('all');
        }else{
            console.log('iframemanager: loading all iframes');
            manager.acceptService('all');
        }
    },

    languages: {
        'en': {
            consent_modal: {
                title: 'Hello traveller, it\'s cookie time!',
                description: 'Our website uses essential cookies to ensure its proper operation and tracking cookies to understand how you interact with it. The latter will be set only after consent. <a href="#privacy-policy" class="cc-link">Privacy policy</a> | <button type="button" data-cc="c-settings" class="cc-link"">Settings</button>',
                primary_btn: {
                    text: 'Accept all',
                    role: 'accept_all'      //'accept_selected' or 'accept_all'
                },
                secondary_btn: {
                    text: 'Reject all',
                    role: 'accept_necessary'       //'settings' or 'accept_necessary'
                },
                revision_message: '<br><br> Dear user, terms and conditions have changed since the last time you visisted!'
            },
            settings_modal: {
                title: 'Cookie settings',
                save_settings_btn: 'Save current selection',
                accept_all_btn: 'Accept all',
                reject_all_btn: 'Reject all',
                close_btn_label: 'Close',
                cookie_table_headers: [
                    {col1: 'Name'},
                    {col2: 'Domain'},
                    {col3: 'Expiration'}
                ],
                blocks: [
                    {
                        title: 'Cookie usage',
                        description: LOREM_IPSUM + ' <a href="#" class="cc-link">Privacy Policy</a>.'
                    }, {
                        title: 'Strictly necessary cookies',
                        description: LOREM_IPSUM + LOREM_IPSUM + "<br><br>" + LOREM_IPSUM + LOREM_IPSUM,
                        toggle: {
                            value: 'necessary',
                            enabled: true,
                            readonly: true  //cookie categories with readonly=true are all treated as "necessary cookies"
                        }
                    }, {
                        title: 'Analytics & Performance cookies',
                        description: LOREM_IPSUM,
                        toggle: {
                            value: 'analytics',
                            enabled: false,
                            readonly: false
                        },
                        cookie_table: [
                            {
                                col1: '^_ga',
                                col2: 'yourdomain.com',
                                col3: 'description ...',
                                is_regex: true
                            },
                            {
                                col1: '_gid',
                                col2: 'yourdomain.com',
                                col3: 'description ...',
                            },
                            {
                                col1: 'cc_youtube',
                                col2: 'yourdomain.com',
                                col3: 'Cookie set by iframemanager'
                            }
                        ]
                    }, {
                        title: 'More information',
                        description: LOREM_IPSUM + ' <a class="cc-link" href="https://orestbida.com/contact/">Contact me</a>.',
                    }
                ]
            }
        }
    }
});
```

**Here you can see what I`ve changed. There you can also use simple HTML as I did to add an Settings link to Box, because its no longer accessible. (Changed the Settings Button to Reject all)**
![Bildschirmfoto 2022-05-19 um 17 41 34](https://user-images.githubusercontent.com/78966793/169340978-3344e3f3-05aa-4a84-b561-878210786850.png)


Im also a beginner with this plugin, but if you have further questions I try my best to help. I also learn always something new. But the frontend (cookieconsent-init.js) is quite simple. I also don't know much about JS so I don't touch the backend for now.

**Reply by montelor** [reply]
Awesome!
You are very fast, congratulations on developing so fast.
I hope I understand everything, I will try it immediately and I will let you know.
I thank you again and I send you best regards !!

Montelor

**Reply by dombyte** [reply]
You're welcome. 
I found this Plugin via a German YouTube Video so I got some first impressions on how to use it. 
But after reading the Docs I got much more knowledge about the logic behind it. Normally I never read the Manuals/Docs because im lazy and do try and error instead, but let me tell you it helps quite a lot and I will continue doing it. 
By the way I Published my Blog on Github (itsdom.org) its not finished yet but the cookie consent plugin is nearly fully implemented as I want. So maybe you can find some inspiration there. The plugin is stored under **static/js**.

Dom

**Reply by montelor** [reply]
Great dombyte!
All this means that there are places where it is still possible to collaborate in peace and think that the world is still beautiful because there are people available like you!

Best regards!

Montelor

**Reply by dombyte** [reply]
Thank you very much for your kind words. Yes with being friendly and kind its easier to learn new things but also teach things. And with helpings others I also can improve my skills because I see problems trough the eyes from someone else.  
Now I need to learn git to also contribute to projects like this. 

Best regards 
Dom

**Reply by dombyte** [reply]
And just a quick reminder, be aware with the situation in your country with using a CDN. Here in Germany it`s quite complicated so to be save I host all necessary files for this plugin by myself and don't use the CDN wich is provided by the Project.

**Comment by orestbida** [comment]
The upcoming v3 version will have a layout with a big "x" icon (accept-necessary only/reject all). Preview: https://user-images.githubusercontent.com/34487268/162553059-2392e1e6-166c-4b0f-9182-db7e1bcee6e3.png

**Reply by tonyC0der** [reply]
Hi all, hi @orestbida ,
I looked around into the v3 versione pre-release source code and demo stuff. I'd like to ask you if it is expected the modal X button (to close modal and reject all cookies) available with bar mode too. From the playground settings I've seen it is available only with box layout.

I'd like to know if the X button is scheduled for v2 version too or not.

Many thanks in advance!

TC

**Reply by orestbida** [reply]
Unfortunately this option is exclusive to v3 (v2 is in maintenance mode and won't receive new feature updates).

The X button is just an alternative to the "reject all" button. The main reason why it is not present in other layouts is because it simply does not fit well from the point of view of the UI.

**Reply by tonyC0der** [reply]
Thank you for your reply! I asked for it because Italian Data Protection Authority guide lines (here a summary in English language [https://secureprivacy.ai/blog/italian-dpa-cookie-guidelines](url)) required "X" button as mandatory feature.
Is it possible to consider an additional choice (in the wide bar) like the one in the attached image?
![image](https://user-images.githubusercontent.com/119496900/205081077-309326ba-7163-4343-82d4-04af2cb5eca6.png)

Thanks again e congrats for this wonderful work!

TC
